src/shared/loop-util.c

```diff
@@ -133,7 +133,19 @@ static int loop_configure_verify(int fd, const struct loop_config *c) {
                         return -errno;
 
                 if (z != c->info.lo_sizelimit) {
-                        log_debug("LOOP_CONFIGURE is broken, doesn't honour .lo_sizelimit. Falling back to LOOP_SET_STATUS64.");
+                        log_debug("LOOP_CONFIGURE is broken, doesn't honour .info.lo_sizelimit. Falling back to LOOP_SET_STATUS64.");
+                        broken = true;
+                }
+        }
+
+        if (c->block_size != 0) {
+                uint32_t z;
+
+                if (ioctl(fd, BLKSSZGET, &z) < 0)
+                        return -errno;
+
+                if (z != c->block_size) {
+                        log_debug("LOOP_CONFIGURE is broken, doesn't honour .block_size. Falling back to LOOP_SET_BLOCK_SIZE.");
                         broken = true;
```

Comment:
actually, I don't think we should consider the ioctl broken here. it definitely makes sense to log about the discrepancy, but neither do we know that any kernels implemented the block size setting incorrectly (which is different for the partscan and the sizelimit stuff), nor is it a really that bad if the block size cannot be set. Yes, it deserves being mentioned, but mostly we should be able to just continue.

hence, drop the `broken = true` here. But keep the debug log message in.

Reply:
How should a failure to set the block size of the loop device be handled at the LUKS volume and user record? e.g. if I specify `--luks-sector-size=4096`, and the `ioctl` fails and sets a 512B block size on the loop device, would a sector size of 512B be stored in the record? (and it should then use a 512B sector size for the LUKS volume and fs).

Reply:
It handle things gracefully. As I understand it's probably fine if LUKS idea of a sector size is larger than the underlying loopback block device's, but not the other way round.

Hence, I'd tell LUKS about the maximum value these two values

1. The value configured in the user record
2. What BLKSSZGET returns from the block device

As far as I understand this should always work then.